### PR TITLE
Added query route tests

### PR DESCRIPTION
**Pull request by c3ho**
## Issue This PR Addresses
Part of #1160 

## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [x] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description
I added a test for the query route. The only issue I'm having replicating mocking is having the actual search portion of the route fail.

You can test the PR with `npm test query`

## Checklist

- [x] **Quality**: This PR builds and passes our npm test and works locally
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

File: test/query.test.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+const request = require('supertest');
+const app = require('../src/backend/web/app');
+
+jest.mock('../src/backend/utils/elastic');
+
+describe('Testing query route', () => {
+  test('Testing with valid characters', async () => {
+    const searchTerm = 'I Love Telescope';
```
Comment (humphd) [inline test/query.test.js:8]:
```suggestion
    const searchTerm = encodeURIComponent('I Love Telescope');
```

File: test/query.test.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    const res = await request(app).get(`/query?search=${searchTerm}`);
+    expect(res.status).toBe(200);
+  });
+
+  test('Testing with invalid characters', async () => {
+    // searchTerm is 257 chars long
+    const searchTerm =
+      '8l4XOYWZ3SA9aevIozTcEAng3GOSCAiiDARThEkAFn2F2YtBexA3lcg1O38SGSHILQrrNYReKWOC6RM4ZQQIGqZoLSOLlbbYqlfSkIDM83aeGDYW7KU8OSLbIXUIWIF4TINwrjxi453biwyjgYsJeqFx9ORd0EIw3dMwGPWhoMbvTIxUWXV032qgPRmohLbTf8xnMyttPjIOk3rHBXpukWSnkZiKyBMsUniZZnxYPw7yIhfoaS77jIPRUuiQufDdO';
+    const res = await request(app).get(`/query?search=${searchTerm}`);
+    expect(res.status).toBe(400);
```
Comment (humphd) [inline test/query.test.js:18]: Let's assert that the length is what we want here so it doesn't get changed in follow-up code (not sure what number you are trying to test exactly).  Also, these aren't "invalid characters" are they?  This test seems to be about maximum length.

```
expect(searchTerm.length).toBeGreaterThan(256);
```

**Review by humphd — CHANGES_REQUESTED** [review]

**Review by humphd — APPROVED** [review]

**Review by manekenpix — APPROVED** [review]
